mldsa/sign.h

```diff
@@ -168,7 +168,15 @@ int crypto_sign(uint8_t *sm, size_t *smlen, const uint8_t *m, size_t mlen,
 int crypto_sign_verify_internal(const uint8_t *sig, size_t siglen,
                                 const uint8_t *m, size_t mlen,
                                 const uint8_t *pre, size_t prelen,
-                                const uint8_t *pk, int externalmu);
+                                const uint8_t *pk, int externalmu)
+__contract__(
+  requires(memory_no_alias(sig, siglen))
+  requires(memory_no_alias(m, mlen))
+  requires(externalmu == 0 || mlen == MLDSA_CRHBYTES)
```

Comment:
Are these "internal" functions allowed to be called by users/clients?

If so, then I'd prefer a more explicit pre-condition that says
"externalmu == 0 || externalmu == 1"
just to be sure.

If they are NOT supposed to be called externally, why are they declared in sign.h? Could they be local, static functions in sign.c? (Happy to move this discussion and any action arising to a new, distinct PR so as not to block this one.)

Reply:
I created #354 for discussion. For now I added `externalmu == 0 || externalmu == 1`.